lib/core/modules/coderunner/codeRunner.js

```diff
@@ -80,11 +85,16 @@ class CodeRunner {
       this.ipc.broadcast("runcode:newCommand", {code});
     }
 
-    if (result instanceof Promise && !result.on) {
-      return result.then((value) => { cb(null, value); }).catch(cb);
+    if (!awaiting) {
```

Comment:
only resolve promise if the code has been wrap and do not consider other promises